### DISCUSSION TITLE
Cannot import from a type-only package

### DISCUSSION THREAD

**Discussion by FelixGirard** [vitest-dev/vitest#6791] [Q&A] [answered]
### Describe the bug

I have a type-only package. We are importing from it and building successfully. 
When vitest encounter that imports, it fails with the error: 
`Failed to resolve entry for package "@types/packageName". The package may have incorrect main/module/exports specified in its package.json.`

### Reproduction

Here is a small reproduction using `@types/base64-js`:

https://stackblitz.com/edit/vitest-dev-vitest-tcivys?file=test%2Fbasic.test.ts

### System Info

```shell
System:
    OS: Windows 11 10.0.22631
    CPU: (20) x64 12th Gen Intel(R) Core(TM) i7-12800H
    Memory: 10.15 GB / 31.67 GB
  Binaries:
    Node: 18.20.2 - C:\Program Files\nodejs\node.EXE
    npm: 10.5.0 - C:\Program Files\nodejs\npm.CMD
    pnpm: 9.12.1 - C:\Program Files\nodejs\pnpm.CMD
  Browsers:
    Edge: Chromium (127.0.2651.74)
    Internet Explorer: 11.0.22621.3527
  npmPackages:
    @vitest/coverage-istanbul: 2.1.2
    vitest: 2.1.2
```

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by hi-ogawa** [comment, marked as answer]
I'm not sure the intent but probably it shouldn't be necessary to import `@types/...` package. Either using `compilerOptions.types` in tsconfig or tsc should already pick that typing automatically.

Btw, node also throws for `import '@types/base64-js'` and it's not a Vitest issue. This is more like a typescript question, so probably it's better to be asked elsewhere.

**Reply by FelixGirard** [reply]
Thank you for your answer. 
It was an internal package, but I'm going to modify it and use the compilerOptions instead.
